src/deadline/job_attachments/asset_sync.py

```diff
@@ -105,6 +126,333 @@ def __init__(
+        self,
+        session_dir: Path,
+        s3_settings: JobAttachmentS3Settings,
+        queue_id: str,
+        job_id: str,
+        attachments: Attachments,
+        step_dependencies: Optional[list[str]] = None,
+        dynamic_mapping_rules: dict[str, PathMappingRule] = {},
+        storage_profiles_path_mapping_rules: dict[str, str] = {},
+    ) -> DefaultDict[str, list[BaseAssetManifest]]:
+        """
+         Args:
+            session_dir: the directory that the session is going to use.
+            s3_settings: S3-specific Job Attachment settings.
+            queue_id:    the ID of the queue for step-step dependency.
+            job_id:      the ID of the job for step-step dependency.
+            attachments: an object that holds all input assets for the job.
+            step_dependencies: the list of Step IDs whose output should be downloaded over the input job attachments.
+            dynamic_mapping_rules: manifest root path to worker host destination mapping relative to local session.
+            storage_profiles_path_mapping_rules: manifest root path to worker host destination mapping given storage profile.
+        Returns: a dictionary of manifest files stored in the session directory.
+        """
+        grouped_manifests_by_root: DefaultDict[str, list[BaseAssetManifest]] = DefaultDict(list)
+
+        for manifest_properties in attachments.manifests:
+            local_root: str = AssetSync.get_local_destination(
+                manifest_properties=manifest_properties,
+                dynamic_mapping_rules=dynamic_mapping_rules,
+                storage_profiles_path_mapping_rules=storage_profiles_path_mapping_rules,
+            )
+
+            if manifest_properties.inputManifestPath:
+                manifest_s3_key = s3_settings.add_root_and_manifest_folder_prefix(
+                    manifest_properties.inputManifestPath
+                )
+                manifest = get_manifest_from_s3(
+                    manifest_key=manifest_s3_key,
+                    s3_bucket=s3_settings.s3BucketName,
+                    session=self.session,
+                )
+                grouped_manifests_by_root[local_root].append(manifest)
+
+        # Handle step-step dependencies.
+        if step_dependencies:
+            for step_id in step_dependencies:
+                manifests_by_root = get_output_manifests_by_asset_root(
+                    s3_settings,
+                    self.farm_id,
+                    queue_id,
+                    job_id,
+                    step_id=step_id,
+                    session=self.session,
+                )
+                for root, manifests in manifests_by_root.items():
+                    dir_name = _get_unique_dest_dir_name(root)
+                    local_root = str(session_dir.joinpath(dir_name))
+                    grouped_manifests_by_root[local_root].extend(manifests)
+
+        return grouped_manifests_by_root
+
+    def launch_vfs(
```

Comment:
`_launch_vfs` since this should be an internal method.

General nit: add `_` for internal methods that is not an API.

Reply:
Yup adding `_`.